frontend/components/ExplainButton.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import Draggable from 'react-draggable';
+import Tooltip from './ToolTip'
+
+const ExplainButton = () => {
+  const [selectedText, setSelectedText] = useState('');
+  const [explanation, setExplanation] = useState('');
+  const [showModal, setShowModal] = useState(false);
+  const [openaiKey, setOpenaiKey] = useState('');
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  useEffect(() => {
+    const fetchOpenAIKey = async () => {
+      try {
+        const response = await fetch('http://localhost:4000/api/openai-key');
```

Comment:
can this be re-written to having the url as a variable or come from the env file, then the endpoint be added to it.

const url = "http://localhost:4000"

```javascript
const response = await fetch(`{url}/api/openai-key`);
```